pkg/resources/manifests/deploy.yaml

```diff
@@ -106,6 +106,8 @@ spec:
       serviceAccount: gadget
       hostPID: true
       hostNetwork: true
+      nodeSelector:
+        kubernetes.io/os: "linux"
```

Comment:
I decided to tackle this now. I added some logic to remove the label filter in case the server version is v1.14.0. It seems to me this logic is simple enough and would allow users running older version to use Inspektor Gadget. What do you think?